File: references/cellab_devkit-1.0 (1)/rules_java/earthgas.java
/*

    Dummy rule definition to invoke user-supplied own code.
    No lookup table is used.

*/

class Earthgas extends ruletable {
    void jcruleModes() {
        setWorld(13);           // Own code, torus world
        setPatternRequest("fullearth");
        setPaletteRequest("fullearth");
        setOwnCodeRequest("randgas");  // User own code function
    }

    int jcrule(int oldstate) {
        return 0;               // No lookup table
    }
}

public class earthgas {
    public static void main(String args[]) {
        (new Earthgas()).generateRuleFile("earthgas");
    }
}

File: references/cellab_devkit-1.0 (1)/ruledefs/earthgas.js
    /*

            Diffuse the Earth with Gravity Turned Off
            
    */
    rule.worldtype = 13;
    rule.patreq = "fullearth";
    rule.palreq = "fullearth";
    rule.ocodereq = "randgas";
    
    function earthgas(oldstate,     nw, n  , ne,
                                    w, self, e,
                                    sw, s  , se) {
        return 0;               // Lookup table is not used
    }
